core/src/cluster_info.rs

```diff
@@ -1951,6 +1939,62 @@ impl ClusterInfo {
         Some(packets)
     }
 
+    fn handle_batch_pull_responses(
+        &self,
+        responses: Vec<(Pubkey, Vec<CrdsValue>)>,
+        thread_pool: &ThreadPool,
+        stakes: &HashMap<Pubkey, u64>,
+        epoch_time_ms: u64,
+    ) {
+        type HashMap = std::collections::HashMap<Pubkey, Vec<CrdsValue>>;
+        if responses.is_empty() {
+            return;
+        }
+        fn extend(hash_map: &mut HashMap, (from, mut data): (Pubkey, Vec<CrdsValue>)) {
+            match hash_map.entry(from) {
```

Comment:
I think in these cases it's more idiomatic to use `if let Some(value) = hash_map.get(&from) {..} else {..}`